src/argilla/utils/dependency.py

```diff
@@ -63,22 +63,29 @@ def _compare_versions(
         )
 
 
-def require_version(requirement: str, func_name: Optional[str] = None) -> None:
+def require_version(requirement: Union[str, List[str]], func_name: Optional[str] = None) -> None:
```

Comment:
```suggestion
def require_version(requirements: Union[str, List[str]], func_name: Optional[str] = None) -> None:
```

Reply:
If the `fn_name` is not widely used along the codebase, I guess we can even separate it to a different decorator, as that's not the most common scenario, and under the condition of `require_version(requirements=["a", "b", "c", "d"], fn_name=["fn_a", None, None, "fn_d"])` we should define a matching list which is not ideal. So I prefer you to detach this functionality and the arg itself from this decorator and create a separate one named `require_dependency_function` or something similar

Reply:
As you guessed, the `fn_name` is not widely used along the  codebase.

- I reviewed the uses of `require_version` and none is making use of `fn_name`.
- With the *decorator* version (`requires_version`) is a bit different as it just takes the `requirements` argument, and grabs the `fn_name` internally from the `func.__name__` attribute:

```python
    def decorator(func: Callable[_P, _R]) -> Callable[_P, _R]:
        @functools.wraps(func)
        def wrapper(*args: _P.args, **kwargs: _P.kwargs) -> _R:
            require_dependencies(requirements, func.__name__)
            return func(*args, **kwargs)

        return wrapper

    return decorator

```

So I think the current implementation assumes that when using the decorator version, it just applies to a single function, like in the following example (it corresponds to the method `score` in `Snorkel` [class](https://github.com/argilla-io/argilla/blob/develop/src/argilla/labeling/text_classification/label_models.py#L609C1-L615C1). 

```python
...
    @requires_version("scikit-learn")
    def score(
        self,
        tie_break_policy: Union[TieBreakPolicy, str] = "abstain",
        output_str: bool = False,
    ) -> Union[Dict[str, float], str]:
        """Returns some scores/metrics of the label model with respect to the annotated records.
...
```

I think that for those cases (the majority), the implementation as is works just fine taking a list of requirements, and a single `fn_name` instead of a list. 
The decorator is also used on `Snorkel` class instead of a method or function, but I think it's just the same logic that for a function.

I think that it could work using a list of `requirements` and a single `fn_name` for the current uses along the codebase, and the current implementation would still be available (single `requirement` with a `fn_name` for the help message), but I understand it may be confusing. 
Anyway, I could detach this to a new `require_dependency_function` but I could use some help with for example the expected use :)